examples/search_examples/server/my_strategy.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ISearchStrategy, PluginStart } from '../../../src/plugins/data/server';
+import { IMyStrategyResponse } from '../common';
+
+export const mySearchStrategyProvider = (data: PluginStart): ISearchStrategy => {
+  const es = data.search.getSearchStrategy('es');
```

Comment:
Wouldn't you just use the `search` function exposed by the service?

Reply:
I just realized we're not exposing the `cancel` function. We should probably be exposing that just like we are with `search`.

Reply:
I wanted this to be an explicit example of consuming a search strategy by name.
I think it's nicer within a strategy, consuming lower level APIs